src/Interpreters/executeQuery.cpp

```diff
@@ -1125,4 +1126,32 @@ void executeQuery(
     streams.onFinish();
 }
 
+void executeTrivialBlockIO(BlockIO & streams, ContextPtr context)
+{
+    try
+    {
+        if (streams.out)
+            throw Exception(ErrorCodes::LOGICAL_ERROR, "Query stream requires input, but no input buffer provided, it's a bug");
```

Comment:
Exception messages does not looks intuitive, maybe worth describing it a little bit?

Reply:
I think it's ok for logical error that contains `it's a bug` in exception message.